docs/usage/private-modules.md

```diff
@@ -5,6 +5,120 @@ description: How to support private modules when using Renovate
+
+In addition to the above options to match against a host, you need to add the credentials. Typically they are either `token`, or `username` + `password`. Other credential terms are not supported yet.
+
+Here is an example of some host rules:
+
+```json
+{
+  "hostRules": [
+    {
+      "hostName": "registry.npmjs.org",
+      "token": "abc123"
+    },
+    {
+      "baseUrl": "https://registry.company.com/pypi-simple/",
+      "username": "engineering",
+      "password": "abc123"
+    }
+  ]
+}
+```
+
+Renovate applies theses `hostRules` to every HTTP(s) request which is sent, so they are largely independent of any platform or datasource logic. With `hostRules` in place, private package lookups should all work.
+
+## Release Notes
+
+When Renovate creates Pull Requests, its default behavior is to locate and embed release notes/changelogs of packages. These release notes are fetched from the source repository of packages and not from the registries themselves, so if they are private then they will require different credentials.
+
+When it comes to open source, most packages host their source on `github.com` in public repositories. However, GitHub greatly rate limits unauthenticated API requests so there is a need to configure credentials for github.com as otherwise the bot will get rate limited quickly. It can be confusing for people who host their own source code privately to be asked to configure a `github.com` token but without it Release Notes for most open source packages will be blocked.
+
+Currently the preferred way to configure `github.com` credentials for self-hosted Renovate is:
+
+- Create a read-only Personal Access Token (PAT) for a `github.com` account. It can be any account, and may even be best to be an empty account created just for this purpose.
+- Add the PAT to Renovate using the environment variable `GITHUB_COM_TOKEN`
+
+## Package Manager Credentials for Artifact Updating
+
+In Renovate terminology, "artifacts" includes lock files, checksum files, and vendored dependencies. One way of understanding artifacts is: "everything else that needs to be updated when the dependency version changes".
+
+Not all package managers supported by Renovate require artifact updating, because not all use lock or checksum files. But when such files need updating, Renovate does so by using the package managers themselves instead of trying to "reverse engineer" each package manager's file formats and behavior. Importantly, such package managers are run via shell commands and do not understand Renovate's `hostRules` objects, so Renovate needs to reformat the credentials into formats (such as environment variables or configuration files) which the package manager understands.
```

Comment:
```suggestion
Not all package managers supported by Renovate require artifact updating, because not all use lock or checksum files.
But when such files need updating, Renovate does so by using the package managers themselves instead of trying to "reverse engineer" each package manager's file formats and behavior.
Importantly, such package managers are run via shell commands and do not understand Renovate's `hostRules` objects, so Renovate needs to reformat the credentials into formats (such as environment variables or configuration files) which the package manager understands.
```